### DISCUSSION TITLE
HomematicKeyMaticIP HmIP-DLD: Incorrect status when using unlock/lock and open door in combination

### DISCUSSION THREAD

**Discussion by Stiefelhagen** [thkl/hap-homematic#456] [Q&A] [answered]
Hi

The new HomematicKeyMaticIP HmIP-DLD is working fine. I can lock und unlock the door, and I can open it.
I configured unlock mode as unlock and checked the "add switch for opening".
There is one caveat:
When the door is locked and I open the door with the opening switch the command is properly executed, but the unlock switch remains in position locked instead of switching to unlocked. So this is a wrong status. For locking I have first to switch to unlock and than to lock back. This is a workaround. 

Do I need to configure something more to live without the workaround, or does this need some code change?

Best regards

Wilhelm

**Comment by thkl** [comment, marked as answer]
This sounds like a bug. I will take a look into this.

**Reply by thkl** [reply]
https://github.com/thkl/hap-homematic/issues/460

**Reply by Stiefelhagen** [reply]
I just checked all combinations. Seems to work correctly. I will close the issue.
